src/dataAccessLayer.ts

```diff
@@ -32,18 +35,46 @@ export class DataAccessLayer {
         return await this.blockchainRepository.getBlockchainAsync();
     }
 
+    public async getBlocksByPeriodAsync(beginTimestamp: number, endTimestamp: number): Promise<Block[]> {
```

Comment:
It would be better to rename this method to:
```ts
public async getBlocksBy"what_kind_of_peroid"Async(beginTimestamp: number, endTimestamp: number)
```
> now it is not clear what period means.

And also a timestamp would not be a good query parameter as a timestamp is too specific. I would suggest using just the date. In that case, the method name would become:
```ts
public async getBlocksByDatePeriodAsync(beginTimestamp: number, endTimestamp: number)
```